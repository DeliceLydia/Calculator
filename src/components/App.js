import { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
// eslint-disable-next-line no-unused-vars
import calculate from '../logic/calculate';

const App = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);
  return (
    <div>
      <>
        <Display />
        <ButtonPanel />
      </>
    </div>
  );
};

export default App;
