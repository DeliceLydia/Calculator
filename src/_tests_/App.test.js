import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../components/App';

test('renders correctly', () => {
  const tree = renderer
    .create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('display numbers', () => {
  render(<App />);
  const button = screen.getByText('9');
  expect(button).toBeInTheDocument();
});

test('displays operators', () => {
  render(<App />);
  const button = screen.getByText(/\+$/i);
  expect(button).toBeInTheDocument();
});

test('display result of calculation', () => {
  render(<App />);
  fireEvent.click(screen.getByText('2'));
  fireEvent.click(screen.getByText('+'));
  fireEvent.click(screen.getByText('8'));
  fireEvent.click(screen.getByText('='));
  const result = screen.getByText('10');
  expect(result).toBeInTheDocument();
});
