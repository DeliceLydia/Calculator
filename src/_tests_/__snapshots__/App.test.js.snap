// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="component-app"
>
  <div
    className="component-display"
  >
    0
  </div>
  <div
    className="button-list"
  >
    <div
      className="groupClass"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn operation"
        onClick={[Function]}
        type="button"
      >
        /
      </button>
    </div>
    <div
      className="groupClass"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn operation"
        onClick={[Function]}
        type="button"
      >
        *
      </button>
    </div>
    <div
      className="groupClass"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn operation"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="groupClass"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn operation"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="groupClass"
    >
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn operation"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
